## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 457952`
Commit: `580849f08776931a4374b1156b9bf4ace31228e8`

---
### `aarch64-darwin`
<details>
  <summary>:x: 12 packages failed to build:</summary>
  <ul>
    <li>nemorosa</li>
    <li>nemorosa.dist</li>
    <li>python312Packages.aiobtclientapi</li>
    <li>python312Packages.aiobtclientapi.dist</li>
    <li>python312Packages.torf</li>
    <li>python312Packages.torf.dist</li>
    <li>python313Packages.aiobtclientapi</li>
    <li>python313Packages.aiobtclientapi.dist</li>
    <li>python313Packages.torf</li>
    <li>python313Packages.torf.dist</li>
    <li>upsies</li>
    <li>upsies.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python312Packages.torf</summary>
<pre>                            &quot;On macOS, try disabling the &#x27;AirPlay Receiver&#x27; service&quot;
                            &quot; from System Preferences -&gt; General -&gt; AirDrop &amp; Handoff.&quot;,
                            file=sys.stderr,
                        )
    
&gt;               sys.exit(1)
E               SystemExit: 1

/nix/store/cajpww8chhgvm8hpc62rxarf5fhw1cgj-python3.12-werkzeug-3.1.3/lib/python3.12/site-packages/werkzeug/serving.py:782: SystemExit
---------------------------- Captured stderr setup -----------------------------
nodename nor servname provided, or not known
=========================== short test summary info ============================
ERROR tests/test_utils.py::test_download_from_unconnectable_url - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_utils.py::test_download_from_connectable_url - SystemExit: 1
ERROR tests/test_utils.py::test_download_with_zero_timeout - SystemExit: 1
ERROR tests/test_magnet.py::test_getting_info__xs_succeeds__as_fails - SystemExit: 1
ERROR tests/test_magnet.py::test_getting_info__xs_fails__as_succeeds - SystemExit: 1
ERROR tests/test_magnet.py::test_getting_info_from_ws - SystemExit: 1
ERROR tests/test_magnet.py::test_getting_info_from_tr - SystemExit: 1
======================= 6715 passed, 7 errors in 21.52s ========================</pre>
</details>
<details>
<summary>python313Packages.torf</summary>
<pre>                            &quot;On macOS, try disabling the &#x27;AirPlay Receiver&#x27; service&quot;
                            &quot; from System Preferences -&gt; General -&gt; AirDrop &amp; Handoff.&quot;,
                            file=sys.stderr,
                        )
    
&gt;               sys.exit(1)
E               SystemExit: 1

/nix/store/piaj774py4hss30i9kc0cqds2ixnbwx5-python3.13-werkzeug-3.1.3/lib/python3.13/site-packages/werkzeug/serving.py:782: SystemExit
---------------------------- Captured stderr setup -----------------------------
nodename nor servname provided, or not known
=========================== short test summary info ============================
ERROR tests/test_utils.py::test_download_from_unconnectable_url - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_utils.py::test_download_from_connectable_url - SystemExit: 1
ERROR tests/test_utils.py::test_download_with_zero_timeout - SystemExit: 1
ERROR tests/test_magnet.py::test_getting_info__xs_succeeds__as_fails - SystemExit: 1
ERROR tests/test_magnet.py::test_getting_info__xs_fails__as_succeeds - SystemExit: 1
ERROR tests/test_magnet.py::test_getting_info_from_ws - SystemExit: 1
ERROR tests/test_magnet.py::test_getting_info_from_tr - SystemExit: 1
======================= 6715 passed, 7 errors in 18.82s ========================</pre>
</details>
</details>

--- a/pkgs/by-name/up/upsies/package.nix
+++ b/pkgs/by-name/up/upsies/package.nix
@@ -1,5 +1,6 @@
 {
   lib,
+  stdenv,
   fetchFromGitea,
   fetchpatch,
   ffmpeg-headless,
@@ -72,10 +73,32 @@ python3Packages.buildPythonApplication rec {
     ]
     ++ runtimeDeps;
 
+  disabledTests = lib.optionals stdenv.hostPlatform.isDarwin [
+    # Fail during object comparisons on Darwin
+    "test_group"
+    "test_has_commentary"
+    "test_special_case"
+    "test_set_release_info"
+    "test_Query_from_release"
+    # Depend on directory format
+    "test_home_directory_property"
+    # Depends on specific cocdecs not available on Darwin
+    "test_generate_episode_queries"
+    # Assert false == true
+    "test_is_mixed_scene_release"
+    # Fails due to expecting a non-darwin path format
+    "test_search"
+  ];
+
   disabledTestPaths = [
     # DNS resolution errors in the sandbox on some of the tests
     "tests/utils_test/http_test/http_test.py"
     "tests/utils_test/http_test/http_tls_test.py"
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    # Fail due to the different set of codecs on Darwin
+    "tests/utils_test/predbs_test/predbs_integration_test.py"
+    "tests/utils_test/release_info_test.py"
   ];
 
   preCheck = ''
@@ -89,6 +112,8 @@ python3Packages.buildPythonApplication rec {
     ":"
     (lib.makeBinPath runtimeDeps)
   ];
+
+  __darwinAllowLocalNetworking = true;
 
   meta = with lib; {
     description = "Toolkit for collecting, generating, normalizing and sharing video metadata";

--- a/pkgs/development/python-modules/aiobtclientapi/default.nix
+++ b/pkgs/development/python-modules/aiobtclientapi/default.nix
@@ -48,6 +48,11 @@ buildPythonPackage rec {
 
   pythonImportsCheck = [ "aiobtclientapi" ];
 
+  disabledTests = [
+    # Timing-sensitive, e.g. "AssertionError: assert 9 <= 7"
+    "test_Monitor_block_until_timeout"
+  ];
+
   disabledTestPaths = [
     # AttributeError
     "tests/clients_test/rtorrent_test/rtorrent_api_test.py"

--- a/pkgs/development/python-modules/torf/default.nix
+++ b/pkgs/development/python-modules/torf/default.nix
@@ -50,9 +50,16 @@ buildPythonPackage rec {
     "test_getting_info__xs_fails__as_fails"
     "test_getting_info__xs_returns_invalid_bytes"
     "test_getting_info__as_returns_invalid_bytes"
+    "test_file_in_singlefile_torrent_has_wrong_size"
+    "test_file_in_singlefile_torrent_doesnt_exist"
+    # Broken assertion
+    # AssertionError: assert 1000 < 1000
+    "test_callback_raises_exception"
   ];
 
   pythonImportsCheck = [ "torf" ];
+
+  __darwinAllowLocalNetworking = true;
 
   meta = {
     description = "Create, parse and edit torrent files and magnet links";

